src/components/auth/Auth.js

```diff
@@ -0,0 +1,64 @@
+import auth0 from 'auth0-js';
+
+class Auth {
+  constructor() {
+    this.auth0 = new auth0.WebAuth({
+      // the following three lines MUST be updated
+      domain: 'coding-coach.eu.auth0.com',
+      audience: 'https://coding-coach.eu.auth0.com/userinfo',
+      clientID: 'bzQKiL4VQtjla1TeIgizh9YIoTyn01Zk',
+      redirectUri: 'http://localhost:3000/callback',
+      responseType: 'id_token',
+      scope: 'openid profile',
+    });
+
+    this.getProfile = this.getProfile.bind(this);
+    this.handleAuthentication = this.handleAuthentication.bind(this);
+    this.isAuthenticated = this.isAuthenticated.bind(this);
+    this.signIn = this.signIn.bind(this);
+    this.signOut = this.signOut.bind(this);
+  }
+
+  getProfile() {
+    return this.profile;
+  }
+
+  getIdToken() {
+    return this.idToken;
+  }
+
+  isAuthenticated() {
+    return new Date().getTime() < this.expiresAt;
+  }
+
+  signIn() {
+    this.auth0.authorize();
+  }
+
+  handleAuthentication() {
+    return new Promise((resolve, reject) => {
+      this.auth0.parseHash((err, authResult) => {
+        if (err) return reject(err);
+        if (!authResult || !authResult.idToken) {
+          return reject(err);
+        }
+        this.idToken = authResult.idToken;
+        this.profile = authResult.idTokenPayload;
+        // set the time that the id token will expire at
+        this.expiresAt = authResult.idTokenPayload.exp * 1000;
+        resolve();
+      });
+    });
+  }
+
+  signOut() {
+    // clear id token, profile, and expiration
+    this.idToken = null;
+    this.profile = null;
+    this.expiresAt = null;
+  }
+}
+
+const auth0Client = new Auth();
```

Comment:
This isn't set in the component proper, just the module (the file). It won't be initialized more than once. I'd put it closer to the top of the app though.

Reply:
Actually, I misread this file. It's not a React component at all, just a traditional `class` with some utility methods in it.

@swissarmykirpan: Consider renaming this file to something like `auth0client` (since you have it named like that in your imports) and moving it out of the components directory. I usually dump imperative code into something like a `/src/utils` or `/src/api` directory.